integrations/kube-agent-updater/cmd/teleport-kube-agent-updater/main.go

```diff
@@ -166,6 +172,19 @@ func main() {
 		os.Exit(1)
 	}
 
+	// The updater is only designed to work with a single deployment or statefulset.
+	// Pre-fetch the ID if it already exists, so that we can use it to lookup versions.
+	// This logic could be moved into the reconciler, but it would require refactoring
+	// all version getter implementations to accept an ID dynamically in GetVersion.
+	updateID, err := getUpdateID(ctx, mgr, kclient.ObjectKey{
+		Namespace: agentNamespace,
+		Name:      agentName + "-updater",
+	})
+	if err != nil {
+		ctrl.Log.Error(err, "failed to get updater ID, exiting")
+		os.Exit(1)
+	}
+
```

Comment:
Can you add an info log containing both the group and updater ID to make troubleshooting easier?